python/whylogs/core/schema.py

```diff
@@ -76,15 +86,18 @@ def __post_init__(self) -> None:
                 LARGE_CACHE_SIZE_LIMIT,
             )
 
-        if self.types:
-            for col, tpe in self.types.items():
-                self._columns[col] = ColumnSchema(
-                    dtype=tpe, resolver=self.resolvers, type_mapper=self.type_mapper, cfg=self.default_configs
-                )
+        for col, tpe in self.types.items():
+            self._columns[col] = ColumnSchema(
+                dtype=tpe, resolver=self.resolvers, type_mapper=self.type_mapper, cfg=self.default_configs
+            )
 
     def copy(self) -> "DatasetSchema":
         """Returns a new instance of the same underlying schema"""
-        copy = DatasetSchema(**self.__dict__)
+        key_dict = vars(self).copy()
+        key_dict.pop("_columns")
+        keys = key_dict.keys()
+        args = {k: self.__dict__[k] for k in keys if k not in self.types}
```

Comment:
I don't understand why the `self.types` items end up in `self.__dict__`

Reply:
when you're doing a copy why not use import deepcopy or import copy? I believe those module are accessible for use

Reply:
Also are you saying you needed to fix this because they ended up in self._dict__ or they are somehow still getting in there?

Reply:
`deepcopy` might be better...

I don't understand how the items in `self.types` end up in `self.__dict__`. Maybe I don't fully understand `vars()`. Line 99 makes sure the `self.types` items aren't passed as arguments to the `DatasetSchema` constructor, which would complain about them because they're not members...

Reply:
That I knew :)  The same thing happened when I used `self.__dict__` instead of `vars()`.  I just don't understand how/where the `self.types` items get into `self.__dict__`.